elasticsearch_dsl/document_base.py

```diff
@@ -58,11 +64,14 @@ class MyDocument(Document):
         s = s.sort(-MyDocument.name)  # sort by name in descending order
     """
 
-    def __init__(self, name, field):
+    def __init__(self, name: str, field: Field):
         self._name = name
         self._field = field
 
-    def __getattr__(self, attr):
+    # note that the return value type here assumes classes will only be used to
+    # access fields (I haven't found a way to make this type dynamic based on a
+    # decision taken at runtime)
+    def __getattr__(self, attr: str) -> "InstrumentedField":
```

Comment:
I'm unhappy about this one, but I don't think it can be handled properly when only at runtime we'll know which type we are returning.